src/app/modules/main/chat_section/chat_content/messages/module.nim

```diff
@@ -413,14 +417,24 @@ method editMessage*(self: Module, messageId: string, updatedMsg: string) =
   self.controller.editMessage(messageId, updatedMsg)
 
 method onMessageEdited*(self: Module, message: MessageDto) =
+  let itemBeforeChange = self.view.model().getItemWithMessageId(message.id)
+  if(itemBeforeChange.isNil):
+    return
+
+  let mentionedUsersPks = itemBeforeChange.mentionedUsersPks
   let renderedMessageText = self.controller.getRenderedText(message.parsedText)
+
   self.view.model().updateEditedMsg(
     message.id,
     renderedMessageText,
     message.containsContactMentions(),
-    message.links
+    message.links,
+    message.mentionedUsersPks
     )
 
+  # ask service to send SIGNAL_MENTIONED_IN_EDITED_MESSAGE signal if there is a new user's mention
+  self.controller.checkEditedMessageForMentions(self.getChatId(), message, mentionedUsersPks)
+
```

Comment:
according to me you could directly do this processing in the service side after you emit an updated for an edited message. here you are listening to the event that a message was edited and then calling the service again. maybe simplify the flow a little?

Reply:
@Khushboo-dev-cpp Unfortunately, there is one problematic use case with the simplified version:
1. Someone edits message and mention you - you should be notified
2. Someone edits message from 1. but do not add another mention - the edited message still has the old mention.
In case 2. we should not notify. In order to do this we need to know the old message text and new message text.
The service do not know old message (or I do not know about it).
In this pr, the list of mentions ids are kept in the model. So when message is coming we can compare: mentions from old message in the model and mentions from new message from service. This comparison is done on server side and a new signal is emitted.
Uff, I hope it makes sense :)

Reply:
@Khushboo-dev-cpp he just keeps a track of mentions for in a certain message in `mentionedUsersPks`. When that message gets edited he just passes mentions from the old message and new edited message dto and service knows how to compare that and what to do next (whether to emit signal or just ignore mentions if logged in used was already mentioned in that edited message). So from my side all good.